packages/graphql-codegen-cli/src/hooks.ts

```diff
@@ -15,37 +15,6 @@ const DEFAULT_HOOKS: Types.LifecycleHooksDefinition<string[]> = {
   beforeAllFileWrite: [],
 };
 
-function normalizeHooks(
-  _hooks: Partial<Types.LifecycleHooksDefinition>
-): Types.LifecycleHooksDefinition<(string | Types.HookFunction)[]> {
-  const keys = Object.keys({
-    ...DEFAULT_HOOKS,
-    ..._hooks,
-  });
-
-  return keys.reduce((prev: Types.LifecycleHooksDefinition<(string | Types.HookFunction)[]>, hookName: string) => {
-    if (typeof _hooks[hookName] === 'string') {
-      return {
-        ...prev,
-        [hookName]: [_hooks[hookName]] as string[],
-      };
-    }
-    if (typeof _hooks[hookName] === 'function') {
-      return {
-        ...prev,
-        [hookName]: [_hooks[hookName]],
-      };
-    }
-    if (Array.isArray(_hooks[hookName])) {
-      return {
-        ...prev,
-        [hookName]: _hooks[hookName] as string[],
-      };
-    }
-    return prev;
-  }, {} as Types.LifecycleHooksDefinition<(string | Types.HookFunction)[]>);
-}
```

Comment:
let's refactor in seperate PR and keep this just isolated for type changes

Reply:
this code contains some type hacks (e.g. casting functions to strings) so for me it felt better to shop both changes at once.

but if it makes the review easier it's fine for me: https://github.com/dotansimha/graphql-code-generator/pull/8652/commits/b1cb681008d1a8ba15f61efed0c3fe639b6a1dda